File: src/Features/SignalsModel.cs
using System;
using Qml.Net;

namespace Features
{
    [Signal("customSignal", NetVariantType.String)]
    public class SignalsModel
    {
        public void RaiseSignal()
        {
            this.ActivateSignal("customSignal", $"Signal was raised at {DateTime.Now.ToLongTimeString()}");
        }
    }
}

File: src/Features.FSharp/SignalsModel.fs
namespace Features

open System
open Qml.Net

[<Signal("customSignal", NetVariantType.String)>]
type SignalsModel() =
    member this.RaiseSignal() =
        let message = "Signal was raised from F# .NET at " + DateTime.Now.ToLongTimeString()
        this.ActivateSignal("customSignal", message)

